src/index.js

```diff
@@ -34,44 +35,66 @@ let player = {
 
 let inventory = [];
 
+const TOWN_COMMANDS = {
+  inventory: () => checkInv(inventory),
+
+  greet,
+
+  hunt: () => {
+    enemy = hunt();
+    return enemy.msgs.encounter;
+  },
+};
+
+const FIGHT_COMMANDS = {
+  attack: () => {
+    const response = attack(player, enemy, inventory);
+    if (enemy.hp <= 0) {
+      enemy = null;
+    }
+    return response;
+  },
+
+  check: () => check(enemy),
+};
+
+const ALL_COMMANDS = new Set([
+  ...Object.keys(TOWN_COMMANDS),
+  ...Object.keys(FIGHT_COMMANDS),
+]);
```

Comment:
Today you learn about `Set`! Set and `Map` are pretty similar, except `Set` does not associate any value to the key, it just says "yeah I got that key" or "nah don't got that key" when you call `set.has(__)`.

`Set` and array are also very similar, you could totally have done this with an array and `array.includes()`, but Set has more performance at large sizes, due to the searching problem we discussed before. `array.includes()` has no strategy for finding things except for checking elements one by one (the array is not sorted it cannot do the binary search I told you about before), but `Set` is able to do that with direct access (more like making an object `set = { x: true, y: true }` and then `set.has('x')` is like doing `set['x']` -- no searching required.

Reply:
Ohh I like the search strategy that we talked about! I'll do some reading on `Set` and see if I can get accustomed to it. More than anything, I still have to become comfortable with the three dots. From what you describe, I can definitely see the benefits of using `Set` though!